thenewboston_node/business_logic/validators.py

```diff
@@ -86,3 +93,30 @@ def validate_exact_value(subject, value, expected_value):
     with validates(f'{subject} value'):
         if value != expected_value:
             raise ValidationError(upper_first(f'{subject} must be equal to {expected_value}'))
+
+
+def validate_network_address(subject, value):
+    with validates(f'{subject} value'):
+        if not is_network_address(value):
+            raise ValidationError(upper_first(f'{subject} must be a valid network address'))
+
+
+def is_network_address(value):
+    return is_fqdn(value) or is_uri(value) or is_ip_network(value)
+
+
+def is_ip_network(value):
+    try:
+        ipaddress.ip_network(value)
+        return True
+    except ValueError:
+        return False
+
+
+def is_fqdn(value):
+    return fqdn.FQDN(value).is_valid
+
+
+def is_uri(value):
+    uri = uritools.urisplit(value)
```

Comment:
why not https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse ?

Reply:
`uritools` docs states that `urlparse` is not fully RFC 3986 compliant due to backward compatibility but I didn't find which part of it. Changed it to `urlparse` for now.